source/common/http/path_utility.h

```diff
@@ -24,5 +29,21 @@ class PathUtil {
   static absl::string_view removeQueryAndFragment(const absl::string_view path);
 };
 
+using Transformation = std::function<absl::optional<std::string>(absl::string_view)>;
```

Comment:
I think you can move it into the `PathTransformer` class

Reply:
Sure, I was thinking maybe we define a Transformation outside of this class.